pp_netlib/gt_prune.py

```diff
@@ -0,0 +1,293 @@
+        # Get the first clique, and see if it has any members already
+        # contained in the vertex list
+        clique = frozenset(next(cliques))
+        if clique.isdisjoint(reference_indices):
+            reference_indices.add(list(clique)[0])
+
+        # Remove the clique, and prune the resulting subgraph (recursively)
+        subgraph = gt.GraphView(graph, vfilt=[v not in clique for v in graph.vertices()])
+        if subgraph.num_vertices() > 1:
+            get_gt_clique_refs(subgraph, reference_indices)
+        elif subgraph.num_vertices() == 1:
+            reference_indices.add(subgraph.get_vertices()[0])
+    except StopIteration:
+        pass
+
+    return reference_indices
+
+def gt_clique_prune(component, graph, reference_indices, components_list):
+    if gt.openmp_enabled():
+        gt.openmp_set_num_threads(1)
+    sys.setrecursionlimit(3000)
+    subgraph = gt.GraphView(graph, vfilt=components_list == component)
+    refs = reference_indices.copy()
+    if subgraph.num_vertices() <= 2:
+        refs.add(subgraph.get_vertices()[0])
+        ref_list = refs
+    else:
+        ref_list = get_gt_clique_refs(subgraph, refs)
+
+    return ref_list
+
+def gt_print_clusters(graph, vertex_labels, backend, out_prefix=None, old_cluster_file=None, external_cluster_csv=None, print_ref=True, print_csv=True, clustering_type="combined", write_unwords=True, use_gpu = False):
+
+    if old_cluster_file == None and print_ref == False:
+        raise RuntimeError("Trying to print query clusters with no query sequences")
+    if write_unwords and not print_csv:
+        write_unwords = False
+
+    # get a sorted list of component assignments
+    if use_gpu:
+        raise NotImplementedError("GPU graph not yet implemented")
+        # use_gpu = check_and_set_gpu(use_gpu, gpu_lib, quit_on_fail = True)
+        # component_assignments = cugraph.components.connectivity.connected_components(G)
+        # component_frequencies = component_assignments['labels'].value_counts(sort = True, ascending = False)
+        # newClusters = [set() for rank in range(component_frequencies.size)]
+        # for isolate_index, isolate_name in enumerate(rlist): # assume sorted at the moment
+        #     component = component_assignments['labels'].iloc[isolate_index].item()
+        #     component_rank_bool = component_frequencies.index == component
+        #     component_rank = np.argmax(component_rank_bool.to_array())
+        #     newClusters[component_rank].add(isolate_name)
+    else:
+        # if backend == "GT":
+        component_assignments, component_frequencies = gt.label_components(graph)
+        component_assignments = list(component_assignments)
+        component_frequencies = list(component_frequencies)
+
+        # elif backend == "NX":
+        #     component_assignments = list((nx.get_node_attributes(graph, "comp_membership")).values())
+        #     component_frequencies = list(len(c) for c in nx.connected_components(graph))
```

Comment:
Clean up these comments. The function is `gt_print_clusters` so only gt is expected to be supported

Reply:
done